src/transformers/models/fnet/modeling_fnet.py

```diff
@@ -166,26 +166,54 @@ def __init__(self, config):
         self._init_fourier_transform(config)
 
     def _init_fourier_transform(self, config):
-        if not config.use_tpu_fourier_optimizations:
-            self.fourier_transform = partial(torch.fft.fftn, dim=(1, 2))
-        elif config.max_position_embeddings <= 4096:
-            if is_scipy_available():
-                self.register_buffer(
-                    "dft_mat_hidden", torch.tensor(linalg.dft(config.hidden_size), dtype=torch.complex64)
-                )
-                self.register_buffer(
-                    "dft_mat_seq", torch.tensor(linalg.dft(config.tpu_short_seq_length), dtype=torch.complex64)
-                )
-                self.fourier_transform = partial(
-                    two_dim_matmul, matrix_dim_one=self.dft_mat_seq, matrix_dim_two=self.dft_mat_hidden
-                )
+        if config.use_latest:
+            if config.use_fft:
+                if self.config.actual_seq_length > 4096 and not math.log2(self.config.actual_seq_length).is_integer():
+                    raise ValueError(
+                        f"For larger seuence lengths (>4096), the actual sequence length {self.config.actual_seq_length} must be a power of 2."
+                    )
+                self.fourier_transform = partial(torch.fft.fftn, dim=(1, 2))
```

Comment:
How recent is this? Should there be a PyTorch version check?

Reply:
@sgugger Do you mean `torch.fft.fftn`?

Reply:
@sgugger The fftn function [was added in PyTorch 1.7](https://github.com/pytorch/pytorch/releases/tag/v1.7.0), in this [PR](https://github.com/pytorch/pytorch/pull/44550).

Reply:
Think a version check would indeed be very nice here. @gchhablani - could you add it here: https://github.com/huggingface/transformers/blob/41436d3dfb98e0d17f018db29790b65663358edf/src/transformers/optimization.py#L301